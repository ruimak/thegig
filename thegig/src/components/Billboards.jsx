import React, { Component } from 'react'
import {getBillboardCharts} from '../api'

export default class Billboards extends Component {
    state = {
        charts:[]
    }

    componentDidMount() {
    this.handleClick = this.handleClick.bind(this);
    }

    handleClick(keyWord) {
       getBillboardCharts(keyWord).then(tracks=>{
            this.setState({charts : tracks.data})
        })
}
   
  render() {
        const listOfButtons = [['Hot100', 'hot-100'], ['RnB', 'rnb'], ['Top UK', 'uksongs'], ['Top Rock', 'rock'], ['Top Pop', 'pop'] ]
const chartButton = (name, keyWord) => {
    return <div>
        <button name={name} onClick={()=>this.handleClick(keyWord)}>
        {name}
        </button> 
        {this.state.charts !== null ?
        <div>
        {this.state.charts.map(track => {
             console.log(track,'dddddddd')
                 return <div>{`Rank: ${track.rank} Track:${track.title}Artist: ${track.artist} ${track.cover}` }</div>
             })}</div> 
             : 
             null}
        </div>
}

    return (
      <div>
             {listOfButtons.map((element, index)=>{
                 return chartButton(listOfButtons[index][0], listOfButtons[index][1])
             }) }
      </div>
    )
  }
}
