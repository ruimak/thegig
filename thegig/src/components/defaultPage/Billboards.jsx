import React, { Component } from 'react'
import {getBillboardCharts} from '../../api'
import { Link } from 'react-router-dom';

export default class Billboards extends Component {
    state = {
        charts:[]
    }

    componentDidMount() {
    this.handleClick = this.handleClick.bind(this);
    }

    handleClick(keyWord) {
       getBillboardCharts(keyWord).then(tracks=>{
            this.setState({charts : tracks.data})
        })
}
   
  render() {
        const listOfButtons = [['Hot100', 'hot-100'], ['RnB', 'rnb'], ['Top UK', 'uksongs'], ['Top Rock', 'rock'], ['Top Pop', 'pop'] ]
const chartButton = (name, keyWord) => {
    return <div>
        <button name={name} onClick={()=>this.handleClick(keyWord)}>
        {name}
        </button> 
       
        </div>
}

    return (
      <div>
             {listOfButtons.map((element, index)=>{
                 return chartButton(listOfButtons[index][0], listOfButtons[index][1])
             }) }
              {this.state.charts !== null ?
        <div>
        {this.state.charts.map(track => {     
                 return  <Link to={`/${track.artist}/song/${track.title}`}><div>  {`Rank: ${track.rank} Track:${track.title}Artist: ${track.artist} ${track.cover}` }</div>
            </Link> })}  </div> 
             : 
             null}
      </div>
    )
  }
}
