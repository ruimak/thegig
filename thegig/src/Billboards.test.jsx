import Billboards from './components/defaultPage/Billboards'
test('Fake Test',() => {
     
    expect(true).toBeTruthy()
})

it("renders correctly", () => {
    
    const wrapper = shallow(
        <Billboards/>
    )
    
    expect(wrapper).toMatchSnapshot()
})

it("renders correctly", () => {
    
    const wrapper = render(
        <Billboards />
        
    )
    
    expect(wrapper).toMatchSnapshot()
})

// t('renders three <Foo /> components', () => {
//     const wrapper = shallow(<Billboards />);
//     expect(wrapper.find('Button')).to.have.lengthOf(3);
//   });i
// describe('List tests', () => {

//     it('renders list-items', () => {
//       const items = ['one', 'two', 'three'];
//       const wrapper = shallow(<List items={items} />);
  
//       // Expect the wrapper object to be defined
//       expect(wrapper.find('.list-items')).toBeDefined();
//       expect(wrapper.find('.item')).toHaveLength(items.length);
//     });
  

it("renders correctly", () => {
    const wrapper = shallow(
        <Billboards/>
    )
    
    // const text = wrapper.find('Paper').to.have.lengthOf(0)
    // // expect(wrapper.find('arrowDown').text()).toEqual('Disabled')
    // expect(text).
})