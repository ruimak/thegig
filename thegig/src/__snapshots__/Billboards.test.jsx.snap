// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Billboards
  classes={
    Object {
      "arrowDown": "Billboards-arrowDown-7",
      "arrowUp": "Billboards-arrowUp-6",
      "artist": "Billboards-artist-12",
      "button": "Billboards-button-1",
      "buttonText": "Billboards-buttonText-2",
      "flatLine": "Billboards-flatLine-9",
      "link": "Billboards-link-3",
      "paper": "Billboards-paper-5",
      "paperSongs": "Billboards-paperSongs-4",
      "picture": "Billboards-picture-11",
      "rank": "Billboards-rank-8",
      "title": "Billboards-title-10",
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<div
  class="Billboards-button-1"
>
  <div
    class="MuiGrid-container-13 MuiGrid-item-14 MuiGrid-grid-xs-12-53"
  >
    <div
      class="MuiGrid-item-14 MuiGrid-grid-xs-2-43"
    >
      <div>
        <button
          class="MuiButtonBase-root-136 MuiButton-root-110 MuiButton-text-112 MuiButton-flat-115 Billboards-buttonText-2"
          name="Hot100"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-111"
          >
            Hot100
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-item-14 MuiGrid-grid-xs-2-43"
    >
      <div>
        <button
          class="MuiButtonBase-root-136 MuiButton-root-110 MuiButton-text-112 MuiButton-flat-115 Billboards-buttonText-2"
          name="RnB"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-111"
          >
            RnB
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-item-14 MuiGrid-grid-xs-2-43"
    >
      <div>
        <button
          class="MuiButtonBase-root-136 MuiButton-root-110 MuiButton-text-112 MuiButton-flat-115 Billboards-buttonText-2"
          name="Top UK"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-111"
          >
            Top UK
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-item-14 MuiGrid-grid-xs-2-43"
    >
      <div>
        <button
          class="MuiButtonBase-root-136 MuiButton-root-110 MuiButton-text-112 MuiButton-flat-115 Billboards-buttonText-2"
          name="Top Rock"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-111"
          >
            Top Rock
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-item-14 MuiGrid-grid-xs-2-43"
    >
      <div>
        <button
          class="MuiButtonBase-root-136 MuiButton-root-110 MuiButton-text-112 MuiButton-flat-115 Billboards-buttonText-2"
          name="Top Pop"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-111"
          >
            Top Pop
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-139 MuiPaper-elevation2-143 MuiPaper-rounded-140 Billboards-paper-5"
  >
      
  </div>
</div>
`;
